backend/app/modules/snapshots/pending/controller.py

```diff
@@ -14,3 +15,12 @@ def create_pending_epoch_snapshot() -> Optional[int]:
         return None
     services: PrePendingServices = get_services(EpochState.PRE_PENDING)
     return services.pending_snapshots_service.create_pending_epoch_snapshot(context)
+
+
+def simulate_pending_epoch_snapshot() -> PendingSnapshotDTO | None:
+    context = state_context(EpochState.PENDING)
```

Comment:
The `context` for it should be `CURRENT`, we want to simulate it before epoch is in pending state. But when we change it it's gonna fail in this line: https://github.com/golemfoundation/octant/blob/develop/backend/app/modules/staking/proceeds/service/aggregated.py#L34
because `end_block` is `None` (which is reasonable, because `end_block` is in the future). 
Please, think how we can resolve this issue. Maybe we should set it to the current block number when we run the simulation?

Reply:
Sure, I went with the same strategy as passing with_blocks flag but for the simulated current epoch. Please take a look :v: